docs/skill.md

```diff
@@ -1,11 +1,81 @@
-## How?
+An agent developer writes skill code that the framework can call.
+
+When you add a skill with the cli, a directory is created which includes modules for the `Behaviour,` `Task`, and `Handler` classes.
+
+
+
+
+## Context
+
+The skill has a `context` object which is shared by all `Handler`, `Behaviour`, and `Task` objects. The skill context has a link to the agent context also.
+
+This means it is possible to, at any point, grab the `context` and have access to the code in other parts of the skill.
+
+For example, in the `ErrorHandler(Handler)` class, the code often grabs a reference to its context and by doing so can access initialised and running framework objects such as a `MailBox` for putting messages into.
+
+``` python
+self.context.outbox.put_message(to=envelope.sender, sender=self.context.agent_public_key,protocol_id=DefaultMessage.protocol_id, message=DefaultSerializer().encode(reply))
+``` 
+
+## What to code
+
+Each of the skill classes has two methods that must be implemented. All of them include a `teardown()` method which the developer must implement. Then there is a specific method for each class that the framework requires.
+
+### `handler.py`
+
+At the current time, each skill has exactly one `Handler` class.
```

Comment:
It is now `handlers.py` and there can be several or none of the `Handler` class. Same for the behaviours and tasks,